src/Core/Services/Implementations/OrganizationService.cs

```diff
@@ -607,23 +668,18 @@ public async Task VerifyBankAsync(Guid organizationId, int amount1, int amount2)
     public async Task<Tuple<Organization, OrganizationUser>> SignUpAsync(OrganizationSignup signup,
         bool provider = false)
     {
-        var plan = StaticStore.PasswordManagerPlans.FirstOrDefault(p => p.Type == signup.Plan);
-        if (plan is not { LegacyYear: null })
-        {
-            throw new BadRequestException("Invalid plan selected.");
-        }
-
-        if (plan.Disabled)
+        var plans = StaticStore.Plans.Where(p => p.Type == signup.Plan).ToList();
+        foreach (var plan in plans)
         {
-            throw new BadRequestException("Plan not found.");
+            ValidateOrganizationUpgradeParameters(plan, signup, signup.UseSecretsManager);
```

Comment:
If you're only calling this on the plans you actually need, you don't need `signup.UseSecretsManager`.